butler.rst

```diff
@@ -149,18 +152,12 @@ Python API
 
 .. py:class:: ButlerConfiguration
 
-    .. py:attribute:: collection
-
-        The :ref:`CollectionTag <Collection>` of the input collection.
-
-    .. py:attribute:: run
-
-        The :ref:`Run` instance used for all outputs.
-
-        May be ``None`` to construct a read-only Butler.
+    A class that maps directly onto a ``YAML`` file.
 
-        The :ref:`Run's <Run>` :ref:`Collection` is always used as the input collection when a :ref:`Run` is provided.
+      * Configuration options are accessed through dictionary keys separated by dots (e.g. ``config['datastore.root']``).
+      * Configuration for :ref:`Datastore` and :ref:`Registry`, including which classes to instantiate, is nested under ``config['datastore']`` and ``config['registry']`` respectively.
 
-    .. py:attribute:: templates
+    .. note::
+        
+        This is an implementation detail that is likely to change significantly.
```

Comment:
In that case, I think we probably shouldn't update the design document, except to remove anything that is no longer true and replace it with a "To Do" (etc) note as needed.